javaparser-core/src/main/java/com/github/javaparser/ast/expr/Name.java

```diff
@@ -238,4 +238,23 @@ public boolean replace(Node node, Node replacementNode) {
         }
         return super.replace(node, replacementNode);
     }
+
+    /**
+     * A top level name is a name that is not contained in a larger Name instance.
+     */
+    public boolean isTopLevel() {
+        return !isInternal();
+    }
+
+    /**
+     * An internal name is a name that constitutes a part of a larger Name instance.
+     */
+    public boolean isInternal() {
+        return getParentNode()
+                .filter(parent -> parent instanceof Name)
+                .map(parent -> (Name) parent)
+                .flatMap(Name::getQualifier)
+                .map(parentNameQualifier -> parentNameQualifier == this)
```

Comment:
Is this check needed? Isn't this always the case?

Reply:
Well, currently Name can have just one position where the child can be another Name (the qualifier position). However it could possibly change in the future. But yes, we could simplify it and just check if the parent is a Name

Reply:
Oh, yes, `Name` can only be qualified by another `Name`. I think it's a safe assumption to make :-)